### DISCUSSION TITLE
How can I disable automatic deletion of lines, trailing spaces and automatic indent upon save?

### DISCUSSION THREAD

**Discussion by rudfol** [LazyVim/LazyVim#586] [Q&A] [answered]
Hi there, 

Here I attach a gif that explains what happens upon save. I would like to keep the indentation I've wrote and the keep more than one empty line between codes. Furthermore it would be great to disable warning for things like '* Line with post space.'
![1 shoot](https://user-images.githubusercontent.com/6250738/232243912-9e51d2ab-16ed-48b5-ab09-6e1ad18f93a3.gif)

In prestige of your work,

Joon

**Comment by amaanq** [comment]
this comes from lua_ls, disable it temporarily with <leader>uf or maybe configure lua_ls, not sure about the latter

**Comment by dpetka2001** [comment, marked as answer]
For line with postspace you could try
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        lua_ls = {
          settings = {
            Lua = {
              diagnostics = {
                disable = { "trailing-space" },
              },
            },
          },
        },
      },
    },
  },
}
```
Haven't tested it but pretty sure that's what triggers the warning about line with post space. About formatting, which is also responsible for identation, you could check the available options [here](https://github.com/LuaLS/lua-language-server/wiki/Formatter), but as the previous post mentioned you could just temporarily disable autoformat on save with `uf` which seems more practical.

**Reply by rudfol** [reply]
That code worked flawlessly. Now how would I be able to view all available options for disable = {"func"} under disgnostics option? I've checked the wiki but it doesn't list them all.

Regards,

**Reply by rudfol** [reply]
Sorry nevermind, I've found them.

**Comment by folke** [comment]
Disable autoformat. Check the LazyVim docs

**Reply by robsonpeixoto** [reply]
Even after disable formatter the trailing spaces is being removed.
Any idea of how to disable it?

**Reply by 3ynm** [reply]
I had the same issue as @robsonpeixoto today. It turned out to be an editorconfig configuration on the specific project.

I don't think this answer will be of any use to robson, but it might be useful to someone else.

**Reply by l3d00m** [reply]
I've found this as well: https://vi.stackexchange.com/a/43233/51750

Set `vim.g.editorconfig = false`

**Comment by rudfol** [comment]
Thank you very much guys. I appreciate your help and the pointers. I will try them one by one and let you know how it goes

**Comment by rudfol** [comment]
I'd make every one of your replies as marked answers if I could. Cheers
